core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -271,20 +272,30 @@ public RewriteDataFilesActionResult execute() {
   private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<DataFile> addedDataFiles,
                                 long startingSnapshotId) {
     try {
-      RewriteFiles rewriteFiles = table.newRewrite()
-          .validateFromSnapshot(startingSnapshotId)
-          .rewriteFiles(Sets.newHashSet(deletedDataFiles), Sets.newHashSet(addedDataFiles));
-      commit(rewriteFiles);
+      doReplace(deletedDataFiles, addedDataFiles, startingSnapshotId);
```

Comment:
nit: wondering if it is necessary to extract this into a `doReplace` method. the current method is not long at all and doReplace is only used once here.

Reply:
i'm sorry for no setting  the pr as a draft, the method is for a  unit test

Reply:
I'm not sure if the modules work out correctly for this, but it may make sense to use RewriteDataFilesCommitter (https://github.com/apache/iceberg/blob/a1bd63d56751999ecee89b871992d7bac395fd52/core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java#L57) here and re-use the commit code from the new path.

Reply:
Actually this may be more complicated than just doing the fix the way you have already done it here ... so maybe forget that